cockroachdb/templates/_helpers.tpl

```diff
@@ -51,11 +51,11 @@ Create chart name and version as used by the chart label.
 {{/*
 Create the name of the ServiceAccount to use.
 */}}
-{{- define "cockroachdb.tls.serviceAccount.name" -}}
-{{- if .Values.tls.serviceAccount.create -}}
-    {{- default (include "cockroachdb.fullname" .) .Values.tls.serviceAccount.name -}}
+{{- define "cockroachdb.serviceAccount.name" -}}
+{{- if .Values.statefulset.serviceAccount.create -}}
```

Comment:
The service account is for statefulset only. Hence it is under the statefulset section in values.yaml. The other places it is used because the statefulset service account has to be attached with role/rolebindings to provide permissions.

I believe we should keep it under the statefulset section only because there are other configurations defined for statefulset in that section and would be consistent to keep it there.

We have changed the `define` name so that the same name is used to create the network policy and the same SA is used in the init job.